pulsar-common/src/main/proto/PulsarApi.proto

```diff
@@ -368,6 +387,7 @@ message CommandRedeliverUnacknowledgedMessages {
 
 message CommandSuccess {
 	required uint64 request_id = 1;
+	optional Schema schema = 2;
```

Comment:
what happens if the message schema changes mid topic?

Reply:
The question there is that there can be multiple schemas valid for the topic a given point in time. Eg: when you rollout an upgrade in the producers they will start publishing a new format, but the old producers are still publishing. 

I think when we talked about this, the solution was that the message itself will carry the version of the schema used so that the consumer can distinguish it. One option could be that you get a list of all valid schemas for the topic when subscribing, but there should also be a way for the consumer to retrieve new schema versions as they are updated.

Reply:
Will the whole schema go with all messages? If that's the case, isn't this field here redundant?

Reply:
Only the version will be stored/sent with each message, and that version is a reference to a particular (valid) schema for that topic.

Reply:
ok, in that case, this makes sense. perhaps it should be a repeated rather than an optional though, for the case that there's different versions.

I guess that the schema name must be immutable for a topic then also.

Reply:
The schema name is, for all intents and purposes, the topic name.

Reply:
I would want to assign a schema to multiple topics - perhaps a large number.

Reply:
That is not impossible, but perhaps inconvenient. What is the use case here? Are we talking within the same namespace or across namespaces? It would be possible to add another layer of indirection to the schema and store the name associated with the topic somewhere (not sure where at the moment).

Reply:
@dave2wave I've been thinking about this more. If this is a common use case I think I can make this happen with little upheaval. Let me know if this is a deal breaker for you.

Reply:
I think it would make sense to have a schema at the namespace level (with per-topic override), since we already have a lot of administration tasks at that level.

Reply:
OK. I had originally proposed that but it was eliminated based on some input but I don't recall the specific circumstances.

Reply:
Why not have the schema registry as part of the namespace with one or more schemas (and versions). One could be default and the others can be chosen to override the topic default. If there are no schemas then the namespace has no schema.